New Bootstrap-UI

tamaskan opened the pull request:
Updated JQuery
Included Standard Roboto-Font
Custom Date-Format via http://momentjs.com/
Coin-Icon-Font via https://labs.allienworks.net/icons/cryptocoins/
Added Total Account-Value and Value per Coin
Different collapsible Sections and Filters
Progressbars and visual Min-Coin-Warning
Increased max. Update-Interval to 3600

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [N/A] **I have read CONTRIBUTING.md**
- [N/A] **I fully understand [Github Flow.](https://guides.github.com/introduction/flow/)**
- [X] **My code adheres to the [code style of this project.](https://poloniexlendingbot.readthedocs.io/en/latest/contributing.html)**
- [N/A] **I have updated the documentation in /docs if I have changed the config, arguments, logic in how the bot works, or anything that understandably needs a documentation change.**
- [N/A] **I have updated the config file accordingly if my change requires a new configuration setting or changes an existing one.**
- [N/A] **I have tested the bot with no issues for 24 continuous hours. If issues were experienced, they have been patched and tested again.**

masterxa committed:
New Bootstrap-UI

Updates JQuery
Roboto-Font
Custom Date-Format via moment.js
Coin-Icon-Font via https://labs.allienworks.net/icons/cryptocoins/
Total Account-Value and Value per Coin
Different collapsible Sections and Filters
Progressbars and visual Min-Coin-Warning
Increased max. Update-Interval

rnevet commented:
General comment, use spaces, keep things correctly aligned.

laxdog commented:
Do we need all the binary files? It's generaly a bad idea to commit binaries to git w/o annex or something similar.

tamaskan commented:
I haven't found a hosted version of the cryptocoin-font, but it could be reduced to a single .woff2-file

masterxa committed:
Reduced Font-Files and show earned Coins

Reduced the CryptoCoin-Font to a single .woff2-File
Show the amount of earned Coins since first UI-Start (Stored in
localstorage)

rnevet commented:
'Show the amount of earned Coins since first UI-Start (Stored in
localstorage)'

Not needed, we are going to have complete lending history in this branch:
https://github.com/BitBotFactory/poloniexlendingbot/pull/349

Please correct tabs as it's really hard to review the code this way.

rnevet commented:
Would it make sense to use https://rawgit.com/ for the fonts files, so they are loaded faster?

rnevet commented:
While I find displaying the opened loans interesting, displaying the cancel loan events is useless as this happens on every bot cycle and doesn't actually bring any information.

rnevet commented:
![image](https://cloud.githubusercontent.com/assets/1645822/26470975/ec2624e0-419f-11e7-9369-fe481b44591e.png)

rnevet commented:
The filtering buttons are ignored when a reload happens. :)

rnevet commented:
I like the overall design, however I think many of the features should be discussed. 

In general I would prefer to have first design change and then separate PRs for features... this will also make it easier to review and approve.

masterxa committed:
Tabs->Spaces, Removed Closed-Panel, Filter-Fix, Removed Earnings

tamaskan commented:
The creators of this bot are free to use only the non-controversial parts and open issues for discussions :-)

Evanito commented on www/lendingbot.js:
```diff
@@ -24,15 +24,80 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
+    var totalvalue = 0;
+    
+//Min Coins based on https://github.com/BitBotFactory/poloniexlendingbot/issues/347
+    var items = [['BTC','BTS','CLAM','DASH','DOGE','ETH','FCT','LTC','MAID','STR','XMR','XRP'],[0.00019124,10,10,0,100,0,100,0,10,100,0,100]];
```
Bitcoin min is actually 0.01, in the issue he was demonstrating the different values of the minimums.

tamaskan replied:
Is there a list of these minimums ?

kingo55 commented:
The design changes look really nice!

Hiding a bunch of stuff under accordions makes this design less practical though. I like to be able to view all the stats upon a glance. If I have to tap to open accordions, it takes time and effort. Arguably not much, but enough that it makes the old design more useful at this stage. 

It gets really annoying when the bot refreshes the page and you lose all the accordion states! :(

tamaskan commented:
I can change that behavior, thanks for the feedback :-)

masterxa committed:
Changed Accordion-Behavior

Coindetail-Accordions should now stay open during refresh

masterxa committed:
Added missing JS

masterxa committed:
Experimental Bootstrap-Design WIP

masterxa committed:
Fixes

masterxa committed:
Changes

Needs testing(notifications,logfilter,mincoin), Todo: settings

masterxa committed:
earnings-merge-modifications

masterxa committed:
Fixes, Table Filter, Notifications

masterxa committed:
Merge remote-tracking branch 'refs/remotes/origin/experimental'

masterxa committed:
Increased Regex-Length for longer Rates

masterxa committed:
Add Loader

masterxa committed:
Grid -> Earnings 24h/Total

tamaskan commented:
https://github.com/dutu/poloLender has a clean interface too, if you want a different approach

masterxa committed:
LocalStorage-Fix & Card-Length

rnevet commented:
How about using prettyFloat instead of printFloat ? :)

tamaskan commented:
No problem, the function-description is quite vague ^^

rnevet commented:
Well it prints a pretty float with a specified accuracy.
i.e. accuracy 2:
1.02332341 -> 1.02**34**
4556.1233 -> 456.**12**
1.00 -> 1
1.10 -> 1.1
etc...

LeaderGL commented:
Hi,
with your branch i always receive the "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error and my "Opened Loan" tabs is empty even if i have opened loan.

LeaderGL commented on www/lendingbot.js:
```diff
@@ -24,25 +24,296 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
+    }
+
+    mydatatable.fnClearTable();
+    mydatatable.fnAddData(dataSet);
```
what if "dataSet" is empty?
we can incur in "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error

LeaderGL commented on www/lendingbot.js:
```diff
@@ -24,25 +24,296 @@ var Satoshi = new BTCDisplayUnit("Satoshi", 100000000);
+    }
+
+    logtable.fnClearTable();
+    logtable.fnAddData(logdataSet);
```
what if "logdataSet" is empty?
we can incur in "DataTables warning: table id={id} - Requested unknown parameter '{parameter}' for row {row-index}, column{column-index}`" error

tamaskan commented:
interesting, never experienced this problem. how does your botlog.json look like ?

masterxa committed:
printFloat->prettyFloat

LeaderGL commented:
at fail time my botlog is like that (i've removed my coin amount):
`{"last_status": "Lent: [0.xxxxxxxx BTC @ 0.1462%] ", "last_update": "2017-06-22 22:44:27", "log": ["2017-06-22 22:44:08 Using custom mindailyrate 0.100% for BTC", "2017-06-22 22:44:14 Not lending BTC due to rate below 0.1561%"], "outputCurrency": {"currency": "BTC", "highestBid": "1"}, "raw_data": {"BTC": {"averageLendingRate": "0.1461733644848254994269884326", "lentSum": "0.xxxxxxxx", "maxToLend": "0.xxxxxxxx", "totalCoins": "0.xxxxxxxx"}}}`

masterxa committed:
Added Check for Table-Data

tamaskan commented:
If there is no active Loan, i would run into the Problem again, so i just added a Check :-)
It would be nice if emilam could push the py-Changes, so i can adapt my Code.

rnevet commented:
btw, I made a JSFiddle for prettyFloat
https://jsfiddle.net/mut72zcp/

masterxa committed:
Added Forecast-Graphs to Grid-View

rnevet commented:
@tamaskan how about making this an alternate theme based on:
https://github.com/BitBotFactory/poloniexlendingbot/commit/86c3521f65df2ef41bd41a62c5b34c45c644cd81
?

mchl18 commented:
some friends and i will work on an alternative frontend soon, but probably on my fork. It really needs an overhaul

rnevet commented:
@mchl18 have you seen the option to create alternate themes?

tamaskan commented:
@rnevet sure, that is not a problem.
@mchl18 awesome :-)

mchl18 commented:
also wanna try and write a config interface...wip

mchl18 commented:
@rnevet no i have not. I just fiddled with some css overlay

tamaskan commented:
thought about a configuration-interface, but the main problem is to keep it up to date

tamaskan commented:
great :-) do you need help ?

mchl18 commented:
I havent had time to set it up, wanna meet with some friends to discuss, if I run into questions i will let you know

mchl18 commented:
vue.js seems tempting but idk if it can handle the whole parsing and all. also i wanted to create tiles for the currencies instead of tables to make it more customisable and responsive

tamaskan commented:
never worked with that, but it should be able to handle that.

mchl18 commented:
yeah it looks really slim and easy to set up, much better than angular and simpler than react, but also only js at its core

utdrmac commented:
@tamaskan What's the status on this? There are some other PRs that want to make UI changes too but this is holding that up and will make the merge of this PR harder. Are you pausing this effort for the moment?

tamaskan commented:
yes, i am still working on this. the bot supports multiple page-templates, so i see no reason why another developer can't create a PR with his Idea.

rnevet commented:
@utdrmac as @tamaskan a recent change to the bot was allowing multiple html templates (IMO might be harder to maintain but it is in). 
https://github.com/BitBotFactory/poloniexlendingbot/commit/86c3521f65df2ef41bd41a62c5b34c45c644cd81

masterxa committed:
Merge remote-tracking branch 'refs/remotes/BitBotFactory/master'

masterxa committed:
Table-Template - Work in Progress

Works with
https://github.com/emilam/poloniexlendingbot/commit/8ac76573651855b43ab193dcf46e1e16144ac704

masterxa committed:
Tab to Space

gincarbone commented:
great job

gincarbone commented:
if someone want to contribute to a project of a trading bot that operates on GDAX (coinbase) take a look at this: https://github.com/gincarbone/roy

bkuhl commented:
Seems like a while since this has had a contribution.  Is this still a work in progress?  Can a screenshot be provided?

bkuhl commented:
Still no updates here.... hrmmmm... what do we do with this PR?

utdrmac commented:
Is this PR dead?

SpiryMedia commented:
I do think it deserves a neat UI. And the initial PR had a very strong idea in that direction.

tamaskan commented:
i switched to coinlend.org, but everyone is free to use the code and create something beautiful
